### PR TITLE
Added `-AsJob` parameter to `Remove-AzVMExtension` cmdlet

### PR DESCRIPTION
**Pull request by pselugar**
## Description

* Added `-AsJob` optional parameter to `Remove-AzVMExtension` cmdlet.
* Added Test case which does the following:
   1) Create a default VM in Azure portal.
   2) Use `Set-AzVMExtension`.
   3) Use `Remove-AzVMExtension`.

## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.cs [hunk 1]
```diff
@@ -234,7 +234,14 @@ public void VirtualMachineGetStatusWithHealhtExtension()
         {
             TestRunner.RunTestScript("Test-VirtualMachineGetStatusWithHealhtExtension");
```
Comment (Sandido) [inline src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.cs:235]: Each ADO bug or work item should have a PR task that has a link to the PR, please add that in all of your bugs.

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1 [hunk 2]
```diff
@@ -3132,6 +3132,66 @@ function Test-VirtualMachineGetStatusWithHealhtExtension
     }
 }
 
+<#
+.SYNOPSIS
+Test Virtual Machines's Status With Health Extension
+Description:
+This test creates a virtual machine and adds a vm health extension.It then removes the health Extension and tests Remove-AzVMExtension
```
Comment (Sandido) [inline src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1:3139]: space after the period.

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1 [hunk 3]
```diff
@@ -3132,6 +3132,66 @@ function Test-VirtualMachineGetStatusWithHealhtExtension
...
+        New-AzResourceGroup -Name $rgname -Location $loc -Force;
+        
+        # VM Profile & Hardware
+        $vmname = 'vm' + $rgname;
+
+        # OS & Image
+        $username = "admin01";
+        $password = $PLACEHOLDER | ConvertTo-SecureString -AsPlainText -Force;
+        $cred = new-object -typename System.Management.Automation.PSCredential -argumentlist $username, $password;
+        [string]$domainNameLabel = "vcrptestps7691-6f2166";
```
Comment (Sandido) [inline src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1:3161]: don't need this [string]

Comment (Sandido) [reply src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1:3161]: don't hardcode the domainnamelabel, rather have something like 'd' + $rgname

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1 [hunk 4]
```diff
@@ -3132,6 +3132,66 @@ function Test-VirtualMachineGetStatusWithHealhtExtension
...
+        # Virtual Machine
+        New-AzVM -ResourceGroupName $rgname -Location $loc -DomainNameLabel $domainNameLabel -Name $vmname -Credential $cred;
+
+        # Adding health extension on VM
+        $publicConfig = @{"protocol" = "http"; "port" = 80; "requestPath" = "/healthEndpoint"};
+        $extensionName = "myHealthExtension"
+        $extensionType = "ApplicationHealthWindows"
+        $publisher = "Microsoft.ManagedServices"
+        Set-AzVMExtension -ResourceGroupName $rgname -VMName $vmname -Publisher $publisher -Settings $publicConfig -ExtensionType $extensionType -ExtensionName $extensionName -Loc $loc -TypeHandlerVersion "1.0"
```
Comment (Sandido) [inline src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1:3170]: all of your lines should have ; at the end of the script statements.

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1 [hunk 5]
```diff
@@ -3132,6 +3132,66 @@ function Test-VirtualMachineGetStatusWithHealhtExtension
...
+
+        # Get VM
+        $vm = Get-AzVM -Name $vmname -ResourceGroupName $rgname -Status;
+
+        # Check for VmHealth Property
+        Assert-NotNull $vm.VMHealth
+        Assert-NotNull $vm.VMHealth.Status
+
+        #Check for VmHealth Extension after removal
```
Comment (Sandido) [inline src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1:3179]: space after the #

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1 [hunk 6]
```diff
@@ -3132,6 +3132,66 @@ function Test-VirtualMachineGetStatusWithHealhtExtension
...
+        $extensionType = "ApplicationHealthWindows"
+        $publisher = "Microsoft.ManagedServices"
+        Set-AzVMExtension -ResourceGroupName $rgname -VMName $vmname -Publisher $publisher -Settings $publicConfig -ExtensionType $extensionType -ExtensionName $extensionName -Loc $loc -TypeHandlerVersion "1.0"
+
+        # Get VM
+        $vm = Get-AzVM -Name $vmname -ResourceGroupName $rgname -Status;
+
+        # Check for VmHealth Property
+        Assert-NotNull $vm.VMHealth
+        Assert-NotNull $vm.VMHealth.Status
```
Comment (Sandido) [inline src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1:3177]: If there is a specific value to check for here, check for that value.

File: src/Compute/Compute/ChangeLog.md [hunk 7]
```diff
@@ -22,6 +22,7 @@
 ## Upcoming Release
 
 ## Version 4.32.0
+* Added `-AsJob` optional parameter for `Remove-AzVMExtension` cmdlet.
```
Comment (Sandido) [inline src/Compute/Compute/ChangeLog.md:25]: move to the Upcoming Release section.

Comment (Sandido) [inline src/Compute/Compute/ChangeLog.md:25]: to the 'Remove-AzVMExtension' cmdlet.

File: src/Compute/Compute/Extension/RemoveAzureVMExtensionCommand.cs [hunk 8]
```diff
@@ -58,6 +58,9 @@ public class RemoveAzureVMExtensionCommand : VirtualMachineExtensionBaseCmdlet
         [ValidateNotNullOrEmpty]
         public SwitchParameter Force { get; set; }
 
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
```
Comment (Sandido) [inline src/Compute/Compute/Extension/RemoveAzureVMExtensionCommand.cs:61]: put each assignment on their own line like the other parameters.

**Review by Sandido — CHANGES_REQUESTED** [review]

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.cs (hunk 1 above)
Comment (pselugar) [reply src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.cs:235]: I tried but this ADO somehow does not have link to PR. Could you please confirm this? there's ADO link in the description

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1 (hunk 2 above)
Comment (pselugar) [reply src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1:3139]: Done

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1 (hunk 3 above)
Comment (pselugar) [reply src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1:3161]: Cool, makes sense

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1 (hunk 6 above)
Comment (pselugar) [reply src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1:3177]: No need for those checks removed them

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1 (hunk 4 above)
Comment (pselugar) [reply src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1:3170]: Good catch!, Done

File: src/Compute/Compute/Extension/RemoveAzureVMExtensionCommand.cs (hunk 8 above)
Comment (pselugar) [reply src/Compute/Compute/Extension/RemoveAzureVMExtensionCommand.cs:61]: Got it

File: src/Compute/Compute/ChangeLog.md (hunk 7 above)
Comment (pselugar) [reply src/Compute/Compute/ChangeLog.md:25]: Done

Comment (pselugar) [reply src/Compute/Compute/ChangeLog.md]: Done

**Comment by VeryEarly** [comment]
please 

- delete ADO workitem link from this PR
- sync your branch with main branch

**Comment by pselugar** [comment]
> please
> 
> * delete ADO workitem link from this PR
> * sync your branch with main branch

Deleted Work Item , also synced branch with main but still unable to undo the merge commit. Could you pls help.

**Comment by VeryEarly** [comment]
> > please
> > 
> > * delete ADO workitem link from this PR
> > * sync your branch with main branch
> 
> Deleted Work Item , also synced branch with main but still unable to undo the merge commit. Could you pls help.

not sure what is the status of your branch, you can try create a new branch from main and cherry-pick all commits from here

**Pull request closed without merging**